common/common.go

```diff
@@ -553,31 +553,31 @@ func ExcludeError(err, excl error) error {
 }
 
 // ErrorCollector allows collecting a stream of errors from concurrent go routines
-// Users should call e.Wg.Done and send errors to e.C
 type ErrorCollector struct {
-	C  chan error
-	Wg sync.WaitGroup
+	errs error
+	wg   sync.WaitGroup
+	m    sync.Mutex
 }
 
-// CollectErrors returns an ErrorCollector with WaitGroup and Channel buffer set to n
-func CollectErrors(n int) *ErrorCollector {
-	e := &ErrorCollector{
-		C: make(chan error, n),
-	}
-	e.Wg.Add(n)
-	return e
+// Collect waits for the internal wait group to be done and returns an error collection
+func (e *ErrorCollector) Collect() (errs error) {
+	e.wg.Wait()
+	e.m.Lock()
+	defer e.m.Unlock()
+	return e.errs
```

Comment:
After collection, you don't clear errors, so it's not reusable. Not huge, but anyone who does retry something will not have any idea that the error they get back is the one from last time